src/geh_common/telemetry/logging_configuration.py

```diff
@@ -11,66 +11,127 @@
+
+        # Instantiate settings (automatically pulls from env vars)
+        logging_settings = LoggingSettings(tracer_name="tracer_name_for_job_name")
+
+        # Configure logging with the settings
+        configure_logging(logging_settings=logging_settings)
+        ```
+    """
+
+    cloud_role_name: str
+    applicationinsights_connection_string: str = Field(repr=False)
+    subsystem: str
+    orchestration_instance_id: UUID | None = None
 
 
 def configure_logging(
     *,
-    cloud_role_name: str,
-    tracer_name: str,
-    applicationinsights_connection_string: str | None = None,
-    extras: dict[str, Any] = {},
-    force_configuration: bool = False,
+    logging_settings: LoggingSettings,
+    extras: dict[str, Any] | None = None,
 ) -> None:
     """Configure logging to use OpenTelemetry and Azure Monitor.
 
-    :param cloud_role_name:
-    :param tracer_name:
-    :param applicationinsights_connection_string:
+    :param logging_settings: Logging settings object
     :param extras: Custom structured logging data to be included in every log message.
-    :param force_configuration: If True, then logging will be reconfigured even if it has already been configured.
     :return:
-
     If connection string is None, then logging will not be sent to Azure Monitor.
     This is useful for unit testing.
     """
     global _TRACER_NAME
-    _TRACER_NAME = tracer_name
+    _TRACER_NAME = logging_settings.cloud_role_name
 
-    # Only configure logging once unless forced.
+    # Only configure logging if not already instrumented
     global _IS_INSTRUMENTED
-    if _IS_INSTRUMENTED and not force_configuration:
+    if _IS_INSTRUMENTED:
         return
 
     # Configure structured logging data to be included in every log message.
-    global _EXTRAS
-    _EXTRAS = extras.copy()
+    if extras is not None:
+        global _EXTRAS
+        _EXTRAS = extras.copy()
 
     # Add cloud role name when logging
-    os.environ["OTEL_SERVICE_NAME"] = cloud_role_name
+    os.environ["OTEL_SERVICE_NAME"] = logging_settings.cloud_role_name
 
     # Configure OpenTelemetry to log to Azure Monitor.
-    if applicationinsights_connection_string is not None:
-        configure_azure_monitor(connection_string=applicationinsights_connection_string)
+    if logging_settings.applicationinsights_connection_string is not None:
+        configure_azure_monitor(connection_string=logging_settings.applicationinsights_connection_string)
         _IS_INSTRUMENTED = True
 
     # Reduce Py4J logging. py4j logs a lot of information.
     logging.getLogger("py4j").setLevel(logging.WARNING)
 
+    # Add extras to log messages
+    if logging_settings.orchestration_instance_id is not None:
+        add_extras({"orchestration_instance_id": str(logging_settings.orchestration_instance_id)})
+    add_extras({"Subsystem": logging_settings.subsystem})
+
+    # Mark logging state as configured
+    global _LOGGING_CONFIGURED
+    _LOGGING_CONFIGURED = True
```

Comment:
Use or remove your function.
```suggestion
    set_logging_configured()
```

Reply:
I use it in conftest.py